File: c_files/raylib/animation_test/src/input.c
#include "input.h"

bool input_map_is_key_pressed(const InputMap *input_map, const Action action)
{
    if (IsKeyPressed(input_map[action].key)) {
        return true;
    }

    return false;
}

bool input_map_is_key_down(const InputMap *input_map, const Action action)
{
    if (IsKeyDown(input_map[action].key)) {
        return true;
    }

    return false;
}

File: python_files/raylib/animation_test/input.py
from enum import IntEnum, auto

import pyray as p


class Action(IntEnum):
    WALK = auto()
    RUN = auto()
    CROUCH = auto()
    JUMP = auto()
    ATTACK = auto()
    SLIDE = auto()
    CLIMB = auto()
    GRAB = auto()
    CAST = auto()


class Input:
    def __init__(self, action: Action, keys: list[p.KeyboardKey]) -> None:
        self.bindings: list[p.KeyboardKey] = keys

    def is_key_down(self, action: Action) -> bool:
        for key in self.bindings:
            if p.is_key_down(key):
                return True

        return False
